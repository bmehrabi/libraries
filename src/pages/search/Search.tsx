import * as React from 'react';
import {createTheme, ThemeProvider} from '@mui/material/styles';
import Container from '@mui/material/Container';
import Grid from '@mui/material/Grid';
import Paper from '@mui/material/Paper';
import Projects from '../../components/project/Projects';
import TextField from '@mui/material/TextField';
import Box from '@mui/material/Box';
import {ChangeEvent, useCallback} from 'react';
import {debounce} from '@mui/material';

const mdTheme = createTheme();

function SearchContent() {
  const [query, setQuery] = React.useState<string>('');
  const searchInputHandler = useCallback((event: ChangeEvent<HTMLTextAreaElement|HTMLInputElement>) => {
    setQuery(event.target.value);
  }, []);

  const debouncedSearchInputHandler = debounce(searchInputHandler, 200);

  return (
    <ThemeProvider theme={mdTheme}>
      <Container maxWidth="lg" sx={{ mt: 4, mb: 4 }}>
        <Grid container spacing={3}>
          {/* Projects table */}
          <Grid item xs={12}>
            <Paper sx={{ p: 2, display: 'flex', flexDirection: 'column' }}>
              <Box sx={{ width: 200 }}>
                <TextField onChange={debouncedSearchInputHandler} label="Search" variant="outlined" />
              </Box>
              <Box sx={{ m: '1rem' }} />
              <Projects query={query} />
            </Paper>
          </Grid>
        </Grid>
      </Container>
    </ThemeProvider>
  );
}

export default function Search() {
  return <SearchContent />;
}
