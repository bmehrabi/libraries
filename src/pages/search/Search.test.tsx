import React from 'react';
import {fireEvent, render, screen, waitFor} from '@testing-library/react';
import Search from './Search';

jest.mock('src/components/project/Projects', () => (props: {query: string}) => {
    return 'Projects: ' + props.query;
});

describe('Search Component', () => {
    it('matches the snapshot', () => {
        const {container} = render(<Search />);

        expect(container).toMatchSnapshot();
    });

    it('uses the search field value to load the projects', async () => {
        render(<Search />);

        const input = screen.getByLabelText(/search/i);
        fireEvent.change(input, {target: {value: 'test'}});

        await waitFor(() => {
            expect(screen.getByText('Projects: test')).toBeInTheDocument();
        });
    });
});


