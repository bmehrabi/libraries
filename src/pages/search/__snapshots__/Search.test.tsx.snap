// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component matches the snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1f2kw6w-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1slnj5u-MuiPaper-root"
        >
          <div
            class="MuiBox-root css-1domaf0"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for=":r0:"
                id=":r0:-label"
              >
                Search
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id=":r0:"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-yjsfm1"
                  >
                    <span>
                      Search
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-tu93bp"
          />
          Projects: 
        </div>
      </div>
    </div>
  </div>
</div>
`;
